pal/audio/operation-queue.ts

```diff
@@ -37,6 +38,38 @@ export interface OperationQueueable {
     _eventTarget: EventTarget;
 }
 
+function removeUnneededCalls (instance: OperationQueueable): void {
+    const size = instance._operationQueue.length;
+    const tmpQueue = instance._operationQueue.slice();
+    const reserveOps: OperationInfo[] = [];
+    let seekSearched = false;
+    for (let i = size - 1; i >= 0; i--) {
+        const opInfo = tmpQueue[i];
+        if (opInfo.op === 'stop') {
+            reserveOps.push(opInfo);
+            break;
+        } else if (opInfo.op === 'seek') {
+            if (!seekSearched) {
+                reserveOps.push(opInfo);
+                seekSearched = true;
+            }
+        } else if (seekSearched) {
+            reserveOps.push(opInfo);
+            break;
+        } else if (reserveOps.length === 0) {
+            reserveOps.push(opInfo);
+        }
+    }
+    reserveOps.forEach((opInfo): void => {
+        const index = instance._operationQueue.indexOf(opInfo);
+        instance._operationQueue.splice(index, 1);
+    });
+    instance._operationQueue.forEach((opInfo): void => {
+        instance._eventTarget.emit(opInfo.id.toString());
```

Comment:
this emit message means the operation is finished, are you sure ?

Reply:
I known, what potential problems will it cause?

Reply:
Currently, only the play interface has a then processing. 
this.node?.emit(AudioSourceEventType.STARTED, this);

Reply:
we only emit the finished message when the operation is finished, see https://github.com/cocos/cocos-engine/blob/f024d3f6a4c86f02d0f9343dcb4a27f30f1dd41d/pal/audio/operation-queue.ts#L46-L52